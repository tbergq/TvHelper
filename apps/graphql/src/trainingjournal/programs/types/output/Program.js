// @flow

import { GraphQLObjectType, GraphQLString } from 'graphql';
import GlobalID from '@adeira/graphql-global-id';
import { GraphQLDate } from 'graphql-iso-date';

export default (new GraphQLObjectType({
  name: 'Program',
  description: 'A trainingjournal program',
  fields: {
    id: GlobalID(({ id }) => id),
    name: {
      type: GraphQLString,
    },
    date: {
      type: GraphQLDate,
    },
  },
}): GraphQLObjectType);
