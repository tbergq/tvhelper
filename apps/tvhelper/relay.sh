#!/bin/bash

cp ../graphql/schema.graphql ./schema.graphql
yarn relay-compiler --language typescript