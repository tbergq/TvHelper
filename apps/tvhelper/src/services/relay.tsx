import { ReactNode } from 'react';
import { Environment, Network, RecordSource, Store, FetchFunction } from 'relay-runtime';
import { RelayEnvironmentProvider } from 'react-relay';
import { RecordMap } from 'relay-runtime/lib/store/RelayStoreTypes';

interface Props {
  readonly children: ReactNode;
  readonly records?: RecordMap;
  readonly token: string | undefined;
}

interface CreateEnvironmentArgs {
  fetchQuery: FetchFunction;
  records?: RecordMap;
}
export const createEnvironment = ({ fetchQuery, records }: CreateEnvironmentArgs) => {
  const network = Network.create(fetchQuery);

  const store = new Store(new RecordSource(records));

  return new Environment({
    network,
    store,
  });
};

const { GRAPHQL_URL } = process.env;

const getBody = (operation, variables) => {
  if (operation.id) {
    return {
      queryId: operation.id,
      variables,
    };
  }
  return {
    query: operation.text,
    variables,
  };
};

export const makeFetchQuery = (token, baseUrl = '') => {
  const fetchQuery = async (operation, variables /* , cacheConfig, uploadables */) => {
    const res = await fetch(`${baseUrl}`, {
      method: 'POST',
      headers: {
        // Add authentication and other headers here
        'content-type': 'application/json',
        'authorization': token,
      },
      body: JSON.stringify(getBody(operation, variables)),
    });
    return res.json();
  };
  return fetchQuery;
};

const EnvironmentProvider = ({ children, records, token }: Props) => {
  const environment = createEnvironment({
    fetchQuery: makeFetchQuery(token, GRAPHQL_URL),
    records,
  });

  return <RelayEnvironmentProvider environment={environment}>{children}</RelayEnvironmentProvider>;
};

export { EnvironmentProvider };
