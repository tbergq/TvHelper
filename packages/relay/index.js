// @flow

export { default as QueryRenderer } from './src/QueryRenderer';
export { graphql, createFragmentContainer } from '@kiwicom/relay';
