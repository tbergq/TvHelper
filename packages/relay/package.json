{
  "name": "@tbergq/relay",
  "private": true,
  "main": "index.js",
  "version": "0.0.1",
  "sideEffects": false,
  "dependencies": {
    "@adeira/fetch": "^1.0.2",
    "@adeira/js": "^1.2.1",
    "@adeira/relay": "^1.3.0",
    "@adeira/relay-utils": "^0.7.0",
    "@tbergq/components": "*",
    "graphql": "^15.0.0",
    "react-fast-compare": "^3.1.1"
  },
  "peerDependencies": {
    "react": "",
    "react-relay": "*",
    "relay-runtime": ""
  },
  "devDependencies": {
    "relay-test-utils": "^9.1.0"
  }
}
