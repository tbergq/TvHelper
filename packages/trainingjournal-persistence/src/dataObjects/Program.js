// @flow strict

import type { ProgramType } from '../models/ProgramModel';

export default class Program {
  id: string;
  name: string;
  date: Date;
  userId: string;

  constructor(program: ProgramType) {
    this.id = program._id.toString();
    this.name = program.name;
    this.date = program.date;
    this.userId = program.user.toString();
  }
}
