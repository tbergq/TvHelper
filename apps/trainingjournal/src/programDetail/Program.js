// @flow

import * as React from 'react';
import { graphql, createFragmentContainer } from '@tbergq/relay';
import { Heading } from '@tbergq/components';
import styled from 'styled-components';
import { defaultTokens } from '@kiwicom/orbit-design-tokens';

import type { Program_program as ProgramType } from './__generated__/Program_program.graphql';
import Weeks from './Weeks';

const WeekList = styled('div')({
  marginTop: defaultTokens.spaceSmall,
});

type Props = {|
  +program: ?ProgramType,
|};

export const Program = (props: Props) => {
  const programName = props.program?.name ?? '';
  return (
    <div>
      <Heading>{programName}</Heading>
      <WeekList>
        <Weeks program={props.program} />
      </WeekList>
    </div>
  );
};

export default createFragmentContainer(Program, {
  program: graphql`
    fragment Program_program on Program {
      name
      ...Weeks_program
    }
  `,
});
