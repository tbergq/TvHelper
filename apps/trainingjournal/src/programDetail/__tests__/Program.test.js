// @flow

import * as React from 'react';
import { QueryRenderer, graphql } from '@kiwicom/relay';
import { createMockEnvironment, MockPayloadGenerator } from 'relay-test-utils';
import { create } from 'react-test-renderer';

import Program from '../Program';

const renderInner = props => <Program program={props.program} />;

test('Program', () => {
  const environment = createMockEnvironment();
  const TestRenderer = () => (
    <QueryRenderer
      environment={environment}
      query={graphql`
        query ProgramQuery @relay_test_operation {
          program(programId: "123") {
            ...Program_program
          }
        }
      `}
      variables={{}}
      onResponse={renderInner}
    />
  );
  const renderer = create(<TestRenderer />);
  environment.mock.resolveMostRecentOperation(operation =>
    MockPayloadGenerator.generate(operation, {
      Program: () => ({ name: 'Program name' }),
    }),
  );

  expect(renderer).toMatchSnapshot();
});
