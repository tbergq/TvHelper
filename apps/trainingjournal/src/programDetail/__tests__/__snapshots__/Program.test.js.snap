// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Program 1`] = `
<div>
  <h1
    className="Heading__StyledHeading-sc-1b8cso5-0 kCqDso"
  >
    Program name
  </h1>
  <div
    className="sc-kgoBCf jUTWkl"
  >
    <div>
      <div
        className="Card__StyledCard-ff0z2v-1 ebswgY"
      >
        <div
          className="Card__StyledChildWrapper-ff0z2v-0 jzawZG"
        >
          <div
            className="CardSection__StyledCardSection-o8zdxr-2 dVGchG"
          >
            <div
              aria-expanded={false}
              className="CardSectionHeader__StyledCardSectionHeader-sc-1gmw3sa-1 fzMVvy"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="0"
            >
              <div
                className="CardSectionHeader__StyledCardSectionHeaderContent-sc-1gmw3sa-3 lcnsxe"
              >
                &lt;mock-value-for-field-"name"&gt;
              </div>
              <svg
                className="CardSectionHeader__StyledCardSectionIconRight-sc-1gmw3sa-0 dtdYCR Icon__StyledIcon-sc-1pnzn3g-0 gbYGej"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
            <div
              className="CardSectionContent__StyledCardSectionContent-sc-1mj8dfx-0 gvncVV"
            >
              <div
                className="Slide__StyledSlide-sc-1xbmdp2-0 dyEvPN"
              >
                <div>
                  <div>
                    <div
                      className="sc-kAzzGY feDzoW"
                    >
                      <div
                        className="sc-cSHVUG fUqPbE"
                      >
                        <h3
                          className="Heading__StyledHeading-sc-1b8cso5-0 kyjUks"
                        >
                          <a
                            href="/programs/day?dayId=<Day-mock-id-2>"
                            onClick={[Function]}
                          >
                            &lt;mock-value-for-field-"name"&gt;
                          </a>
                        </h3>
                        <div>
                          &lt;mock-value-for-field-"name"&gt;
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-chPdSV hYxHKu"
      />
    </div>
  </div>
</div>
`;
